sqlglot/parser.py

```diff
@@ -1255,7 +1248,27 @@ def _parse_escape(self, this):
         return self.expression(exp.Escape, this=this, expression=self._parse_string())
 
     def _parse_bitwise(self):
-        return self._parse_tokens(self._parse_term, self.BITWISE)
+        this = self._parse_term()
+
+        while True:
+            if self._match_set(self.BITWISE):
+                this = self.expression(
+                    self.BITWISE[self._prev.token_type],
+                    this=this,
+                    expression=self._parse_term(),
+                )
+            elif self._match_pair(TokenType.LT, TokenType.LT):
```

Comment:
This change means that I can have sqlglot parse `SELECT 4 << 1` as Spark dialect and if I write it back out as Spark it will convert to `SELECT SHIFTLEFT(4, 1)`. That output is right but that input is invalid Spark. Before I was seeing that as a problem but it seems like sqlglot as a library is ok with this type of invalid input as long as the output is right?

Reply:
correct, i parse a superset of sql